src/network_analysis.jl

```diff
@@ -192,6 +192,122 @@ function complexstoichmat(::Type{Matrix{Int}}, rn::ReactionSystem, rcs)
     Z
 end
 
+"""
+    laplacianmat(rn::ReactionSystem, pmap=Dict(); sparse=false)
+
+    Return the negative of the graph Laplacian of the reaction network. The ODE system of a chemical reaction network can be factorized as ``\frac{dx}{dt} = Y A_k Φ(x)``, where Y is the [`complexstoichmat`](@ref) and A_k is the negative of the graph Laplacian, and Φ is the [`massactionvector`](@ref). A_k is an n-by-n matrix, where n is the number of complexes, where A_{ij} = k_{ij} if a reaction exists between the two complexes and 0 otherwise. 
+    Returns a symbolic matrix by default, but will return a numerical matrix if parameter values are specified via parammap. 
+"""
+function laplacianmat(rn::ReactionSystem, pmap = Dict(); sparse = false) 
+    D = incidencemat(rn; sparse); K = fluxmat(rn, pmap; sparse)
+    D*K
+end
+
+"""
+    fluxmat(rn::ReactionSystem, pmap = Dict(); sparse=false)
+
+    Return an r×c matrix K such that, if complex j is the substrate complex of reaction i, then K_{ij} = k, the rate constant for this reaction. Mostly a helper function for the network Laplacian, [`networklaplacianmat`](@ref). Has the useful property that ``\frac{dx}{dt} = S*K*Φ(x)``, where S is the [`netstoichmat`](@ref) or net stoichiometry matrix and Φ(x) is the [`massactionvector`](@ref).
+    Returns a symbolic matrix by default, but will return a numerical matrix if rate constants are specified as a `Tuple`, `Vector`, or `Dict` of symbol-value pairs via `pmap`.
+"""
+function fluxmat(rn::ReactionSystem, pmap::Dict = Dict(); sparse=false) 
+    rates = reactionrates(rn)
+
+    if !isempty(pmap)
+        length(pmap) != length(parameters(rn)) && error("Incorrect number of parameters were specified.")
+        pmap = symmap_to_varmap(rn, pmap)
+        pmap = Dict(ModelingToolkit.value(k) => v for (k, v) in pmap)
+        rates = [substitute(rate, pmap) for rate in rates]
+    end
```

Comment:
Can this code be extracted to one function that is then called in the various cases for species or parameters in these functions?